presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -964,23 +1245,33 @@ public int hashCode()
         public RootInternalResourceGroup(
                 String name,
                 BiConsumer<InternalResourceGroup, Boolean> jmxExportListener,
-                Executor executor)
+                Executor executor,
+                Integer version)
         {
-            super(Optional.empty(), name, jmxExportListener, executor, true);
+            super(Optional.empty(), name, jmxExportListener, executor, true, version);
         }
 
-        public synchronized void processQueuedQueries()
+        public RootInternalResourceGroup(String name, BiConsumer<InternalResourceGroup, Boolean> jmxExportListener, Executor executor, InternalResourceGroup root, Integer version)
         {
-            internalRefreshStats();
-            while (internalStartNext()) {
-                // start all the queries we can
+            super(Optional.empty(), name, jmxExportListener, executor, root, true, version);
+        }
+
+        public void processQueuedQueries()
+        {
+            synchronized (getRoot()) {
```

Comment:
Why do we need to call `getRoot()` here since it seems to just return root? Does the function call serve any purpose? If not should we remove it to avoid overhead?

Reply:
it's to access the private variable of the parent class, but can access it using super.root so going to replace it.

Reply:
Ah, ok, that makes sense. Either way is fine then. Thanks!